src/main/java/gyro/google/compute/HealthCheckResource.java

```diff
@@ -0,0 +1,475 @@
+ *              request-path: "/myapp"
+ *              response: "okay"
+ *          end
+ *      end
+ *
+ * Advanced TCP Health Check with Custom Port Name
+ * -----------------------------------------------
+ *
+ * .. code-block:: gyro
+ *
+ *      google::compute-health-check health-check-example
+ *          check-interval-sec: 30
+ *          description: "The description goes here."
+ *          healthy-threshold: 8
+ *          name: "tcp-advanced"
+ *          timeout-sec: 29
+ *          type: "TCP"
+ *          unhealthy-threshold: 6
+ *
+ *          tcp-health-check
+ *              port: 501
+ *              port-name: "custom-port"
+ *              proxy-header: "PROXY_V1"
+ *              request-path: "/myapp"
+ *              response: "okay"
+ *          end
+ *      end
+ */
+@Type("compute-health-check")
+public class HealthCheckResource extends ComputeResource implements Copyable<HealthCheck> {
+    private Integer checkIntervalSec;
+    private String description;
+    private HealthCheckHttpHealthCheck httpHealthCheck;
+    private HealthCheckHttpsHealthCheck httpsHealthCheck;
+    private HealthCheckHttp2HealthCheck http2HealthCheck;
+    private HealthCheckSslHealthCheck sslHealthCheck;
+    private HealthCheckTcpHealthCheck tcpHealthCheck;
+    private Integer healthyThreshold;
+    private String name;
+    private String selfLink;
+    private Integer timeoutSec;
+    private String type;
+    private Integer unhealthyThreshold;
+
+    /**
+     * The name of the health check. (Required)
+     */
+    @Id
+    @Required
+    @Regex("(?:(?:[-a-z0-9]{1,63}\\.)*(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?):)?(?:[0-9]{1,19}|(?:[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?))")
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The description of the health check.
+     */
+    @Updatable
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * The time (in seconds) for how often to send a health check.
+     */
+    @Updatable
+    public Integer getCheckIntervalSec() {
+        return checkIntervalSec;
+    }
+
+    public void setCheckIntervalSec(Integer checkIntervalSec) {
+        this.checkIntervalSec = checkIntervalSec;
+    }
+
+    /**
+     * The http Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckHttpHealthCheck
+     */
+    @Updatable
+    public HealthCheckHttpHealthCheck getHttpHealthCheck() {
+        return httpHealthCheck;
+    }
+
+    public void setHttpHealthCheck(HealthCheckHttpHealthCheck httpHealthCheck) {
+        this.httpHealthCheck = httpHealthCheck;
+    }
+
+    /**
+     * The https Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckHttpsHealthCheck
+     */
+    @Updatable
```

Comment:
You should add a `@ConflictsWith` annotation to this because only one can be set at a time:
```suggestion
    @Updatable
    @ConflictsWith({"http-health-check", "http2-health-check", "ssl-health-check", "tcp-health-check"})
```